lib/util/cmd.go

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+package util
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/adobe/aquarium-fish/lib/log"
+)
+
+// Runs & logs the executable command
+func RunAndLog(sec string, timeout time.Duration, stdin io.Reader, path string, arg ...string) (string, string, error) {
+	var stdout, stderr bytes.Buffer
+
+	// Running command with timeout
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, path, arg...)
+
+	log.Debugf("%s: Executing: %s %s", sec, cmd.Path, strings.Join(cmd.Args[1:], " "))
```

Comment:
Is there any concern over secrets getting logged here?  Specifically, things like `docker` credentials?  I looked through the use-cases, I don't _think_ it is a problem.  Just wanted to call out in case it might be.  If it is, it might be easiest to add another parameter `logCall bool` to allow the call-site to disable logging?  Might be a tedious change not worth realizing.

Alternatively, since it's called `RunAndLog`, a different `Run` method that doesn't log could be added.  Then just make a helper method that has this additional `logCalls` parameter, and both `Run` and `RunAndLog` pass-through to this helper function.  It's only necessary if there are locations we are worried about secrets getting dumped to the console, but I couldn't determine if those even exist.

Reply:
If someone runs system as debug - it's their problem, but overall you right, we need to ensure no creds are exposed in the log, but it's a tricky task to solve right away. Created a ticket for that: #88

Reply:
> but it's a tricky task to solve right away

Agreed.  Yeah, this looks good for now, we can deal with it in the future.